x-pack/plugins/security_solution/cypress/tasks/api_calls/elasticsearch.ts

```diff
@@ -16,6 +16,61 @@ export const deleteIndex = (index: string) => {
   });
 };
 
+export const createIndex = (indexName: string, properties: Record<string, unknown>) =>
+  cy.request({
+    method: 'PUT',
+    url: `${Cypress.env('ELASTICSEARCH_URL')}/${indexName}`,
+    headers: { 'kbn-xsrf': 'cypress-creds' },
+    body: {
+      mappings: {
+        properties,
+      },
+    },
+  });
+
+export const indexDocument = (indexName: string, document: Record<string, unknown>) =>
+  cy.request({
+    method: 'POST',
+    url: `${Cypress.env('ELASTICSEARCH_URL')}/${indexName}/_doc`,
+    headers: { 'kbn-xsrf': 'cypress-creds' },
+    body: document,
+  });
+
+export const waitForRulesToFinishExecution = (ruleIds: string[]) => {
```

Comment:
Why can't we rely on checking rule status to determine if it's done executing? We just need to make sure the rules executed once right?

Reply:
That was my idea as well. Or do you mean why not call Kibana's Rules API (`api/detection_engine/rules/_find`) and check execution status value there?

Reply:
I meant using the find API like you said to check status.